src/main/java/com/robinpowered/sdk/RobinApi.java

```diff
@@ -83,6 +83,10 @@ public RobinApi(Credential credential) {
      * @param httpClient An HTTP client for performing network requests.
      */
     public RobinApi(Credential credential, Client httpClient) {
+        if (credential != null && credential.getValue() == null) {
+            throw new IllegalArgumentException("RobinApi credential cannot have a null value");
```

Comment:
Just "The credential cann..." would be fine. You dont need to be self referential